# Study 897 — CPPI Floor — the teardown

The three-book race (CPPI / buy-and-hold / matched static), the excess-of-cash identity, the leverage-clean re-timing spanning alpha, the block-bootstrap Sharpe-difference CI, the two-era cut, the multiplier / floor / cost sweeps, the gap-risk stress, and the 30-seed synthetic control.

In [1]:
R = {'alpha_ann': -1.56,
 'avg_w': 0.38,
 'bear_breach': 0,
 'bear_prot': 0.52,
 'bear_prot_min': 0.11,
 'beta': 0.705,
 'bh_cagr': 10.7,
 'bh_dd': -55.2,
 'bh_sharpe': 0.542,
 'bh_vol': 19.78,
 'boot_hi': -0.05,
 'boot_lo': -0.691,
 'boot_point': -0.379,
 'boot_win': 1.4,
 'calm_prot': 0.005,
 'cost0': -0.379,
 'cost1': -0.394,
 'cost10': -0.554,
 'cost10_dd': -17.3,
 'cost2': -0.413,
 'cost5': -0.472,
 'cppi_cagr': 2.37,
 'cppi_dd': -20.8,
 'cppi_sharpe': 0.164,
 'cppi_vol': 7.94,
 'crash08_bh': -36.8,
 'crash08_bh_dd': -47.1,
 'crash08_c': -11.6,
 'crash08_c_dd': -11.2,
 'crash20_bh': 18.3,
 'crash20_bh_dd': -33.7,
 'crash20_c': -4.8,
 'crash20_c_dd': -16.3,
 'crash22_bh': -18.2,
 'crash22_bh_dd': -24.5,
 'crash22_c': -18.7,
 'crash22_c_dd': -20.8,
 'diff_t_nw': -2.15,
 'end': '2026-06-30',
 'era_e_bh': 0.335,
 'era_e_cppi': -0.464,
 'era_e_n': 2164,
 'era_e_t': -2.38,
 'era_e_vs': -0.799,
 'era_l_bh': 0.759,
 'era_l_cppi': 0.751,
 'era_l_n': 2637,
 'era_l_t': -0.74,
 'era_l_vs': -0.008,
 'fl70_cagr': 4.83,
 'fl70_dd': -31.9,
 'fl70_sh': 0.323,
 'fl80_cagr': 2.37,
 'fl80_dd': -20.8,
 'fl80_sh': 0.164,
 'fl90_cagr': 1.78,
 'fl90_dd': -10.6,
 'fl90_sh': 0.123,
 'floor': 80,
 'fp': '4b5a30d250b4',
 'gap15_fire': 0,
 'gap25_fire': 20,
 'gap_breach_at': 20,
 'gap_thresh': 20,
 'm3_dd': -33.5,
 'm3_sh': 0.517,
 'm3_w': 0.67,
 'm4_dd': -31.4,
 'm4_sh': 0.417,
 'm5_dd': -20.8,
 'm5_sh': 0.164,
 'm6_dd': -18.0,
 'm6_sh': -0.242,
 'm8_dd': -19.2,
 'm8_sh': -0.288,
 'mult': 5,
 'n_breach': 0,
 'n_days': 4801,
 'n_seeds': 30,
 'sharpe_vs_bh': -0.379,
 'start': '2007-05-31',
 'static_cagr': 5.27,
 'static_dd': -23.0,
 'static_sharpe': 0.542,
 'static_vol': 7.42,
 't_alpha': -1.16,
 'turnover': 2.3}

## The race — CPPI vs buy-and-hold vs matched static mix (excess-of-cash, gross)

A *constant* fraction of SPY funded from cash has the **same** excess-of-cash Sharpe as SPY — so the matched-static and buy-and-hold Sharpes coincide, and CPPI's only distinguishing act is the dynamic **re-timing** (the spanning alpha).

In [2]:
print(f"CPPI        : excess-Sharpe {R['cppi_sharpe']:.3f}  CAGR {R['cppi_cagr']:5.2f}%  "
      f"vol {R['cppi_vol']:5.2f}%  maxDD {R['cppi_dd']:.1f}%")
print(f"buy-and-hold: excess-Sharpe {R['bh_sharpe']:.3f}  CAGR {R['bh_cagr']:5.2f}%  "
      f"vol {R['bh_vol']:5.2f}%  maxDD {R['bh_dd']:.1f}%")
print(f"static@avgw : excess-Sharpe {R['static_sharpe']:.3f}  CAGR {R['static_cagr']:5.2f}%  "
      f"vol {R['static_vol']:5.2f}%  maxDD {R['static_dd']:.1f}%")
print(f"vs buy-and-hold {R['sharpe_vs_bh']:+.3f}   re-timing alpha {R['alpha_ann']:+.2f}%/yr "
      f"(HAC t {R['t_alpha']:+.2f}, beta {R['beta']:.3f})   avg-w {R['avg_w']:.2f}, "
      f"turnover {R['turnover']:.1f}x/yr, breaches {R['n_breach']}")

CPPI        : excess-Sharpe 0.164  CAGR  2.37%  vol  7.94%  maxDD -20.8%
buy-and-hold: excess-Sharpe 0.542  CAGR 10.70%  vol 19.78%  maxDD -55.2%
static@avgw : excess-Sharpe 0.542  CAGR  5.27%  vol  7.42%  maxDD -23.0%
vs buy-and-hold -0.379   re-timing alpha -1.56%/yr (HAC t -1.16, beta 0.705)   avg-w 0.38, turnover 2.3x/yr, breaches 0


## Bootstrap — circular block CI on the excess-Sharpe difference (CPPI − buy-and-hold)

In [3]:
print(f"gain {R['boot_point']:+.3f}  95% CI [{R['boot_lo']:+.3f}, {R['boot_hi']:+.3f}]  "
      f"P(CPPI wins) {R['boot_win']:.1f}%   -> the CI sits entirely below zero")

gain -0.379  95% CI [-0.691, -0.050]  P(CPPI wins) 1.4%   -> the CI sits entirely below zero


## Crash years — the floor's best day (2008) and worst (2020)

In [4]:
for yr,bh,bd,c,cd in [(2008,R['crash08_bh'],R['crash08_bh_dd'],R['crash08_c'],R['crash08_c_dd']),
                      (2020,R['crash20_bh'],R['crash20_bh_dd'],R['crash20_c'],R['crash20_c_dd']),
                      (2022,R['crash22_bh'],R['crash22_bh_dd'],R['crash22_c'],R['crash22_c_dd'])]:
    print(f"{yr}: BH {bh:+6.1f}% (DD {bd:6.1f}%)  ->  CPPI {c:+6.1f}% (DD {cd:6.1f}%)")

2008: BH  -36.8% (DD  -47.1%)  ->  CPPI  -11.6% (DD  -11.2%)
2020: BH  +18.3% (DD  -33.7%)  ->  CPPI   -4.8% (DD  -16.3%)
2022: BH  -18.2% (DD  -24.5%)  ->  CPPI  -18.7% (DD  -20.8%)


## Robustness — two eras (split 2016-01-01)

The Sharpe shortfall concentrates in the volatile first half (re-timing alpha significantly negative); the calmer half ties on Sharpe but the drawdown protection *fades* (a cash-accreting floor no longer sits under an appreciated NAV — a TIPP/ratchet floor fixes this).

In [5]:
print(f"2007-2015 (n={R['era_e_n']}): CPPI-Sh {R['era_e_cppi']:+.3f}  BH-Sh {R['era_e_bh']:+.3f}  "
      f"vs-BH {R['era_e_vs']:+.3f}  alpha-t {R['era_e_t']:+.2f}")
print(f"2016-2026 (n={R['era_l_n']}): CPPI-Sh {R['era_l_cppi']:+.3f}  BH-Sh {R['era_l_bh']:+.3f}  "
      f"vs-BH {R['era_l_vs']:+.3f}  alpha-t {R['era_l_t']:+.2f}")

2007-2015 (n=2164): CPPI-Sh -0.464  BH-Sh +0.335  vs-BH -0.799  alpha-t -2.38
2016-2026 (n=2637): CPPI-Sh +0.751  BH-Sh +0.759  vs-BH -0.008  alpha-t -0.74


## The frontier — multiplier & floor sweeps (no setting beats buy-and-hold's 0.542)

In [6]:
print('multiplier:')
for m,sh,dd in [(3,R['m3_sh'],R['m3_dd']),(4,R['m4_sh'],R['m4_dd']),(5,R['m5_sh'],R['m5_dd']),
                (6,R['m6_sh'],R['m6_dd']),(8,R['m8_sh'],R['m8_dd'])]:
    print(f"  m={m} (breach>{100//m}%): CPPI-Sh {sh:+.3f}  maxDD {dd:.1f}%")
print('floor:')
for fl,sh,dd,cg in [(70,R['fl70_sh'],R['fl70_dd'],R['fl70_cagr']),(80,R['fl80_sh'],R['fl80_dd'],R['fl80_cagr']),
                    (90,R['fl90_sh'],R['fl90_dd'],R['fl90_cagr'])]:
    print(f"  floor {fl}%: CPPI-Sh {sh:+.3f}  maxDD {dd:.1f}%  CAGR {cg:.2f}%")

multiplier:
  m=3 (breach>33%): CPPI-Sh +0.517  maxDD -33.5%
  m=4 (breach>25%): CPPI-Sh +0.417  maxDD -31.4%
  m=5 (breach>20%): CPPI-Sh +0.164  maxDD -20.8%
  m=6 (breach>16%): CPPI-Sh -0.242  maxDD -18.0%
  m=8 (breach>12%): CPPI-Sh -0.288  maxDD -19.2%
floor:
  floor 70%: CPPI-Sh +0.323  maxDD -31.9%  CAGR 4.83%
  floor 80%: CPPI-Sh +0.164  maxDD -20.8%  CAGR 2.37%
  floor 90%: CPPI-Sh +0.123  maxDD -10.6%  CAGR 1.78%


## Costed & gap risk — no return edge to erode, and the floor isn't a hard guarantee

CPPI trails buy-and-hold **at zero cost** (not a friction story); and a one-day drop past **1/m = 20%** breaches the floor — daily rebalancing can't pre-empt an overnight gap.

In [7]:
for c,vs in [('0bp',R['cost0']),('1bp',R['cost1']),('2bp',R['cost2']),('5bp',R['cost5']),('10bp',R['cost10'])]:
    print(f"  {c:>4}: CPPI vs buy-and-hold {vs:+.3f}")
print(f"gap risk: a single overnight drop > 1/m = {R['gap_thresh']}% breaches the floor "
      f"(SPY never gapped that far close-to-close in 2007-2026, so the real-tape floor held)")

   0bp: CPPI vs buy-and-hold -0.379
   1bp: CPPI vs buy-and-hold -0.394
   2bp: CPPI vs buy-and-hold -0.413
   5bp: CPPI vs buy-and-hold -0.472
  10bp: CPPI vs buy-and-hold -0.554
gap risk: a single overnight drop > 1/m = 20% breaches the floor (SPY never gapped that far close-to-close in 2007-2026, so the real-tape floor held)


## Synthetic control — the machinery is unbiased (live, offline)

In [8]:
import os, sys
sys.path.insert(0, os.path.abspath('..'))
sys.path.insert(0, os.path.abspath(os.path.join('..','..','..')))
import numpy as np
from cppi import data, strategy as st
bear = np.array([st.synthetic_detect(data.synthetic_prices(seed=897+s, n_days=2500, drift=-0.0004, sigma=0.016)[0])['dd_protection'] for s in range(8)])
calm = np.array([st.synthetic_detect(data.synthetic_prices(seed=897+s, n_days=2500, drift=0.0004, sigma=0.006)[0])['dd_protection'] for s in range(8)])
gap  = np.array([st.synthetic_detect(data.synthetic_prices(seed=897+s, n_days=4000, drift=0.0003, sigma=0.011, gap_prob=0.01, gap_size=0.25)[0])['n_breach'] for s in range(8)])
print(f"bear (8 seeds): drawdown protection mean {bear.mean():+.3f}  (floor holds)")
print(f"calm (8 seeds): drawdown protection mean {calm.mean():+.3f}  (nothing to protect)")
print(f"gap 25%>20% (8 seeds): floor breach fired in {(gap>0).sum()}/8")

bear (8 seeds): drawdown protection mean +0.572  (floor holds)
calm (8 seeds): drawdown protection mean +0.008  (nothing to protect)
gap 25%>20% (8 seeds): floor breach fired in 8/8


## Verdict

- **Signal — MIXED.** The floor-protection claim is **real and mechanical**: maxDD **-20.8%** vs buy-and-hold's **-55.2%**, **0** breaches, 2008 cut from −47% to −11%, confirmed by a 30-seed synthetic control. But the implied *'insurance improves the risk-adjusted outcome'* claim is **false**: excess Sharpe **0.164 vs 0.542** (bootstrap gain -0.379, 95% CI [-0.691, -0.050]), no multiplier or floor beats buy-and-hold, and the re-timing spanning alpha **subtracts** value (-1.56%/yr, *t* = -2.38 in the volatile era). A single-cycle, GFC-anchored, ~19-year window.
- **Tradability — MIRAGE.** No bankable *return* edge — CPPI trails buy-and-hold gross, before costs. The drawdown cap is cheap and genuine **as insurance** (2.3×/yr turnover on SPY, survives every cost) but it is a *cost* paid in full out of upside (2.4% CAGR vs 10.7%), not a paycheck — and an overnight gap past 1/m = 20% breaches even the floor.